securesystemslib/interface.py

```diff
@@ -900,6 +900,260 @@ def import_ecdsa_privatekey_from_file(filepath, password=None):
+    filepath = os.path.join(os.getcwd(), spx_key['keyid'])
+
+  else:
+    logger.debug('The filepath has been specified.  Not using the key\'s'
+        ' KEYID as the default filepath.')
+
+  # Does 'filepath' have the correct format?
+  # Ensure the arguments have the appropriate number of objects and object
+  # types, and that all dict keys are properly named.
+  # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
+  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
+
+  # If the caller does not provide a password argument, prompt for one.
+  if password is None: # pragma: no cover
+
+    # It is safe to specify the full path of 'filepath' in the prompt and not
+    # worry about leaking sensitive information about the key's location.
+    # However, care should be taken when including the full path in exceptions
+    # and log files.
+    password = get_password('Enter a password for the SPX'
+        ' key (' + Fore.RED + filepath + Fore.RESET + '): ',
+        confirm=True)
+
+  else:
+    logger.debug('The password has been specified. Not prompting for one.')
+
+  # Does 'password' have the correct format?
+  securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
+
+  # If the parent directory of filepath does not exist,
+  # create it (and all its parent directories, if necessary).
+  securesystemslib.util.ensure_parent_dir(filepath)
+
+  # Create a temporary file, write the contents of the public key, and move
+  # to final destination.
+  file_object = securesystemslib.util.TempFile()
+
+  # Generate the spx public key file contents in metadata format (i.e.,
+  # does not include the keyid portion).
+  keytype = spx_key['keytype']
+  keyval = spx_key['keyval']
+  scheme = spx_key['scheme']
+  spxkey_metadata_format = securesystemslib.keys.format_keyval_to_metadata(
+      keytype, scheme, keyval, private=False)
+
+  file_object.write(json.dumps(spxkey_metadata_format).encode('utf-8'))
+
+  # Write the public key (i.e., 'public', which is in PEM format) to
+  # '<filepath>.pub'.  (1) Create a temporary file, (2) write the contents of
+  # the public key, and (3) move to final destination.
+  # The temporary file is closed after the final move.
+  file_object.move(filepath + '.pub')
+
+  # Write the encrypted key string, conformant to
+  # 'securesystemslib.formats.ENCRYPTEDKEY_SCHEMA', to '<filepath>'.
+  file_object = securesystemslib.util.TempFile()
+
+  # Encrypt the private key if 'password' is set.
+  if len(password):
+    spx_key = securesystemslib.keys.encrypt_key(spx_key, password)
+
+  else:
+    logger.debug('An empty password was given. '
+                 'Not encrypting the private key.')
+    spx_key = json.dumps(spx_key)
+
+  # Raise 'securesystemslib.exceptions.CryptoError' if 'spx_key' cannot be
+  # encrypted.
+  file_object.write(spx_key.encode('utf-8'))
+  file_object.move(filepath)
+
+  return filepath
+
+
+
+
+def import_spx_publickey_from_file(filepath):
```

Comment:
Not necessary for this PR: this function can be generalized across all keytypes.